middlewares/inputValidation.js

```diff
@@ -0,0 +1,50 @@
+import { check, validationResult } from 'express-validator/check';
+
+const validate = {
+  signup: [
+    check('email')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Email is required')
+      .isEmail()
+      .trim()
+      .withMessage('Please input a valid email address'),
+    check('firstName')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('First name is required')
+      .isAlpha()
+      .trim()
+      .withMessage('First name can only contain letters'),
+    check('lastName')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Last name is required')
+      .isAlpha()
+      .trim()
+      .withMessage('Last name can only contain letters'),
+    check('password')
+      .not()
+      .isEmpty({ ignore_whitespace: true })
+      .withMessage('Password is required')
+      .matches(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z]/, 'i')
+      .withMessage('Password must contain at least one uppercase letter, one lowercase letter and one numeric digit')
+      .trim()
+      .isLength({ min: 8 })
+      .withMessage('Password must be at least 8 characters'),
+    (req, res, next) => {
+      const errors = validationResult(req);
+      const errorMessage = [];
```

Comment:
if you are displaying **one error message** at a time, then it's best you use an object instead of an array.

Reply:
Noted. Thanks